### DISCUSSION TITLE
Integrating ADCP Data into a Simulated HYCOM Velocity Field

### DISCUSSION THREAD

**Discussion by claudioiturra** [Parcels-code/Parcels#1797] [General]
Hello, I'm trying to incorporate a single-point current velocity measurement, obtained from an ADCP, over a simulated HYCOM velocity field. My goal is to evaluate how these single-point measurements interact with the Lagrangian circulation simulated by the HYCOM field. I have already run parcel trajectories over the HYCOM field (attached). Is it possible to somehow incorporate the ADCP data into this analysis
![001](https://github.com/user-attachments/assets/871b26df-9512-4f37-a51e-41d2fe58ebf4)

**Comment by erikvansebille** [comment]
Dear @claudioiturra, I'm not sure this is a question I can help you with; it's more a scientific question ("what behaviour do you want for your particles") than a technical question. But I can imagine you create a Kernel that uses one set of Fields if a particle is close to the ADCP location and another set of Fields otherwise? 

There are many ways to implement this, but the easiest is perhaps to use [`NestedFields`](https://docs.oceanparcels.org/en/latest/examples/tutorial_NestedFields.html), where you then have to create your ADCP Field in such a way that it is only valid for a certain, rectangular, region.

Note by the way that the flickering in your animation is likely because particles that are deleted at times not multiple of `outputdt` also get plotted. To get rid of the flickering (if you would want to), rethink the way you set your `frames` argument in `FuncAnimation()`. See also the last [Animations section of the Parcels Output tutorial](https://docs.oceanparcels.org/en/latest/examples/tutorial_output.html#Animations).
